packages/apps/src/types/plugin/plugin.ts

```diff
@@ -66,6 +66,8 @@ export interface IPlugin<TEvents extends {} = {}> {
    */
   onActivity?(event: IPluginActivityEvent): void | Promise<void>;
 
+  buildActivityContext?(activity: Activity): TCtxAddons;
```

Comment:
would prefer if we kept to the `onXYZ` naming convention to keep things consistent, but we can also maybe reuse the `onActivity` for this and change the return type to `void | TContext | Promise<void | TContext>`